### PR TITLE
Improve rtpsrelaylib coverage

### PR DESCRIPTION
**Pull request by jrw972**

### CODE REVIEW

File: tests/unit-tests/tools/dds/rtpsrelaylib/Name.cpp [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+  Atom character_class(false, abc);
+  size_t h = AtomHash()(character_class);
+  EXPECT_EQ(h, 6291457u);
+}
+
+void test_valid(const std::string& s, const Name& expected, bool pattern)
+{
+  Name actual(s);
+  EXPECT_TRUE(actual.is_valid());
+  EXPECT_TRUE(actual == expected);
```
Comment (iguessthislldo) [inline tests/unit-tests/tools/dds/rtpsrelaylib/Name.cpp:140]:
```suggestion
  EXPECT_EQ(actual, expected);
```

File: tests/unit-tests/tools/dds/rtpsrelaylib/Utility.cpp [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+  ACE_INET_Addr addr("127.0.0.1:1234");
+  AddrPort ap(addr, SPDP);
+  GuidAddr ga1(guid, ap);
+  GuidAddr ga2(guid, ap);
+  EXPECT_EQ(ga1, ga2);
+}
+
+TEST(tools_dds_rtpsrelaylib_Utility, GuidAddr_not_equal)
+{
+  OpenDDS::DCPS::GUID_t guid1;
```
Comment (iguessthislldo) [inline tests/unit-tests/tools/dds/rtpsrelaylib/Utility.cpp:89]:
```suggestion
  OpenDDS::DCPS::GUID_t guid1 = OpenDDS::DCPS::GUID_UNKNOWN;
```

To get rid of the warnings?

**Review by iguessthislldo — CHANGES_REQUESTED** [review]

File: tools/dds/rtpsrelaylib/Name.cpp [hunk 3]
```diff
@@ -105,16 +86,10 @@ char Name::parse_character(Name& name, const std::string& buffer, size_t& idx)
...
-    return Atom(0);
-  }
-
   name.is_pattern_ = true;
 
-  if (buffer[idx++] != '[') {
-    name.is_valid_ = false;
-  }
+  // Skip '[']
+  idx++;
```
Comment (mitza-oci) [inline tools/dds/rtpsrelaylib/Name.cpp:92]:
```suggestion
  ++idx;
```

File: tools/scripts/unit_test_coverage.sh [hunk 4]
```diff
@@ -25,7 +25,17 @@ function get-units {
...
             sort -u
     else
-        echo "${args[@]}"
+        for arg in "${args[@]}"
+        do
+            if [[ -d "${arg}" ]]
+            then
+                find "${arg}" -path tools/rapidjson -prune -false -o -path tools/modeling/tests -prune -false -o -name "*.h" -o -name "*.cpp" -o -name "*.inl" |
+            sed -E -e '/[CS]\.(h|cpp|inl)$/ d' -e '/TypeSupportImpl\.(h|cpp)$/ d' -e '/[Ee]xport.h$/ d' -e 's/\.[^.]*$//' |
+            sort -u
```
Comment (mitza-oci) [inline tools/scripts/unit_test_coverage.sh:34]: Indent level looks off - but not sure what it should be for shell.

Pattern matching on file names to determine generated files may not be accurate.  Alternatives to consider would be running a git command (assuming this is always a repo), or having another mode of the script that would run before `make` and create a text file listing the non-generated source file names.

File: tools/dds/rtpsrelaylib/Utility.h [hunk 5]
```diff
@@ -82,19 +82,39 @@ struct GuidAddr {
...
+      static const OpenDDS::DCPS::GUID_tKeyLessThan gc;
       return gc(guid, other.guid);
     }
     return address < other.address;
   }
 };
 
+inline bool operator==(const EntityId_t& x, const EntityId_t& y)
+{
+  return std::memcmp(&x, &y, sizeof(x)) == 0;
```
Comment (mitza-oci) [inline tools/dds/rtpsrelaylib/Utility.h:94]: `#include <cstring>`

**Review by mitza-oci — CHANGES_REQUESTED** [review]

File: tools/scripts/unit_test_coverage.sh (hunk 4 above)
Comment (iguessthislldo) [reply tools/scripts/unit_test_coverage.sh:34]: I am using `git` to check for ignored files in the [lint script](https://github.com/objectcomputing/OpenDDS/blob/05c82664fc481a2d9cc620d3eb4f00f9911c14ff/tools/scripts/lint.pl#L1059-L1072).

**Review by mitza-oci — APPROVED** [review]

File: tests/unit-tests/tools/dds/rtpsrelaylib/Name.cpp [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+}
+
+std::string stringify(const Atom& atom)
+{
+  std::stringstream ss;
+  ss << atom;
+  return ss.str();
+}
+
+TEST(tools_dds_rtpsrelaylib_Name, Atom_stringify)
```
Comment (iguessthislldo) [inline tests/unit-tests/tools/dds/rtpsrelaylib/Name.cpp:104]: In case you haven't seen it, this test is failing.

**Review by iguessthislldo — APPROVED** [review]
Other than the failing test, it looks alright to me.

**Pull request merged**